simd/performance_tests/include/PerfTest_Operators.hpp

```diff
@@ -0,0 +1,389 @@
+  }
+};
+
+class ceils {
+ public:
+  template <typename T>
+  auto on_host(T const& a) const {
+    return Kokkos::ceil(a);
+  }
+  template <typename T>
+  KOKKOS_INLINE_FUNCTION auto on_device(T const& a) const {
+    return Kokkos::ceil(a);
+  }
+};
+
+class rounds {
+ public:
+  template <typename T>
+  auto on_host(T const& a) const {
+    return Kokkos::round(a);
+  }
+  template <typename T>
+  KOKKOS_INLINE_FUNCTION auto on_device(T const& a) const {
+    return Kokkos::round(a);
+  }
+};
+
+class truncates {
+ public:
+  template <typename T>
+  auto on_host(T const& a) const {
+    return Kokkos::trunc(a);
+  }
+  template <typename T>
+  KOKKOS_INLINE_FUNCTION auto on_device(T const& a) const {
+    return Kokkos::trunc(a);
+  }
+};
+
+class shift_right {
+ public:
+  template <typename T, typename U>
+  auto on_host(T const& a, U const& b) const {
+    return a >> b;
+  }
+  template <typename T, typename U>
+  KOKKOS_INLINE_FUNCTION auto on_device(T const& a, U const& b) const {
+    return a >> b;
+  }
+};
+
+class shift_left {
+ public:
+  template <typename T, typename U>
+  auto on_host(T const& a, U const& b) const {
+    return a << b;
+  }
+  template <typename T, typename U>
+  KOKKOS_INLINE_FUNCTION auto on_device(T const& a, U const& b) const {
+    return a << b;
+  }
+};
+
+class minimum {
+ public:
+  template <typename T>
+  auto on_host(T const& a, T const& b) const {
+    return Kokkos::min(a, b);
+  }
+  template <typename T>
+  KOKKOS_INLINE_FUNCTION auto on_device(T const& a, T const& b) const {
+    return Kokkos::min(a, b);
+  }
+};
+
+class maximum {
+ public:
+  template <typename T>
+  auto on_host(T const& a, T const& b) const {
+    return Kokkos::max(a, b);
+  }
+  template <typename T>
+  KOKKOS_INLINE_FUNCTION auto on_device(T const& a, T const& b) const {
+    return Kokkos::max(a, b);
+  }
+};
+
+#if defined(KOKKOS_ENABLE_DEPRECATED_CODE_4)
+#if defined(KOKKOS_ENABLE_DEPRECATION_WARNINGS)
+KOKKOS_IMPL_DISABLE_DEPRECATED_WARNINGS_PUSH()
+#endif
+
+class hmin {
+ public:
+  template <typename T, typename MaskType>
+  auto on_host(T const& a, MaskType const& mask) const {
+    return Kokkos::Experimental::hmin(where(mask, a));
+  }
+
+  template <typename T, typename MaskType>
+  KOKKOS_INLINE_FUNCTION auto on_device(T const& a,
+                                        MaskType const& mask) const {
+    return Kokkos::Experimental::hmin(where(mask, a));
+  }
+};
+
+class hmax {
+ public:
+  template <typename T, typename MaskType>
+  auto on_host(T const& a, MaskType const& mask) const {
+    return Kokkos::Experimental::hmax(where(mask, a));
+  }
+
+  template <typename T, typename MaskType>
+  KOKKOS_INLINE_FUNCTION auto on_device(T const& a,
+                                        MaskType const& mask) const {
+    return Kokkos::Experimental::hmax(where(mask, a));
+  }
+};
+
+#if defined(KOKKOS_ENABLE_DEPRECATION_WARNINGS)
+KOKKOS_IMPL_DISABLE_DEPRECATED_WARNINGS_POP()
+#endif
+#endif
```

Comment:
I'd argue that we just remove these. These are deprecated in favor of `reduce_min` and `reduce_max` respectively and are effectively using same implementation underneath.